lib/src/presentation/view/splash_screen.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+
+import '../../core/util/numeric_constants.dart';
+import '../../core/util/route_constants.dart';
+import '../../core/util/string_constants.dart';
+import '../widget/splash_title_animation.dart';
+
+class SplashScreen extends StatefulWidget {
```

Comment:
Why do you need an StatefulWidget?

Reply:
Because i need a context to make the navigation after the build of the screen, that's why i do it on the init state. In stateless i can't get the context to do that. Is there another way to get that context or to navigate to the next screen?

Reply:
due to you have the navigation inside of splash screen StatefulWidget kind of make sense, but you are delegating the heavy lifting (loading, preparing the app, doing heavy stuff) to the splash screen.  In your code is simulated with the delay function.

If we think in SOLID principle Single responsibility won't be met here due to Splashscreen is showing a GIF and loading (simulated) resources.

A better approach is to do could be on the home page check if you are set if you show the Splashscreen (not navigate) meanwhile you are loading your data, once is done navigate to your landing page.

Probably at this point is too early to do what I'm suggesting you, but keep in mind Splashscreens are very often a GOD class.

That being said let's keep statefulllwidget for now, but as soon as you start putting more complex logic in the Splashscreen you should refactor this code.

Reply:
Thank you, im going to have this in mind for the next prs